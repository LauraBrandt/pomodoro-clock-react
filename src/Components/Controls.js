import React from 'react';

const Controls = () => {
  return(
    <div>
      <button id="start_stop"><i className="fas fa-play" title="start"></i></button>
      <button id="reset"><i className="fas fa-sync-alt" title="reset"></i></button>
    </div>
  );
}

export default Controls;