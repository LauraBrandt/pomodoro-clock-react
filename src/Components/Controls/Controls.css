.controls {
  display: flex;
  justify-content: center;
}

.controls button {
  color: #f0f0f0;
  font-size: 1.5rem;
  border: 1px solid #eee;
  border-radius: 5px;
  background: #252525;
  padding: 0.3rem 0.8rem;
  width: 2.2em;
  margin: 0.3em 0.8rem 2rem 0.8rem;
  cursor: pointer;
}

.controls button:hover,
.controls button:focus {
  color: white;
  background: #333;
  outline: none;
}

.controls button:active {
  background: #3f3f3f;
}