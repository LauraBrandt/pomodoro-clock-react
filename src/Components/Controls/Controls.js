import React from 'react';
import PropTypes from 'prop-types';
import './Controls.css';

const Controls = ({isRunning, reset, handleStartStop, isMuted, handleMute}) => {
  return(
    <div className="controls">
      <button id="start_stop" title={`${isRunning ? 'Stop' : 'Start'}`} onClick={handleStartStop}>
        <i className={`fas fa-${isRunning ? 'pause' : 'play'}`} title={`${isRunning ? 'Stop' : 'Start'}`}></i>
      </button>
      <button id="reset" title="Reset" onClick={reset}>
        <i className="fas fa-sync-alt" title="reset"></i>
      </button>
      <button id="mute" title={`${isMuted ? 'Unmute' : 'Mute'}`} onClick={handleMute}>
        <i className={`fas fa-${isMuted ? 'volume-off' : 'volume-up'}`} title={`${isMuted ? 'Unmute' : 'Mute'}`}></i>
      </button>
    </div>
  );
}

Controls.propTypes = {
  isRunning: PropTypes.bool.isRequired,
  reset: PropTypes.func.isRequired,
  handleStartStop: PropTypes.func.isRequired,
  isMuted: PropTypes.bool.isRequired,
  handleMute: PropTypes.func.isRequired
}

export default Controls;