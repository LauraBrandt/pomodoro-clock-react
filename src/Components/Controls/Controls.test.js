import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Controls from './Controls';

const props = {
  isRunning: false,
  reset: jest.fn(),
  handleStartStop: jest.fn(),
  isMuted: false,
  handleMute: jest.fn()
}

it('renders without crashing', () => {
  shallow(<Controls {...props}/>);
});

describe('structure', () => {
  it('has all elements with ids necessary', () => {
    const wrapper = shallow(<Controls {...props}/>);
    expect(wrapper.find('#start_stop')).toHaveLength(1);
    expect(wrapper.find('#reset')).toHaveLength(1);
    expect(wrapper.find('.fa-sync-alt')).toHaveLength(1);
    expect(wrapper.find('#mute')).toHaveLength(1);
  });

  it('sets titles and classes correctly based on isRunning prop', () => {
    const wrapper = shallow(<Controls {...props}/>);

    expect(wrapper.find('.fa-pause')).toHaveLength(0);
    expect(wrapper.find('.fa-play')).toHaveLength(1);
    expect(wrapper.find('#start_stop').filter("[title='Start']")).toHaveLength(1);
    expect(wrapper.find('.fas').filter("[title='Start']")).toHaveLength(1);
    
    wrapper.setProps({ isRunning : true });
    
    expect(wrapper.find('.fa-pause')).toHaveLength(1);
    expect(wrapper.find('.fa-play')).toHaveLength(0);
    expect(wrapper.find('#start_stop').filter("[title='Stop']")).toHaveLength(1);
    expect(wrapper.find('.fas').filter("[title='Stop']")).toHaveLength(1);
  });

  it('sets titles and classes correctly based on isMuted prop', () => {
    const wrapper = shallow(<Controls {...props}/>);

    expect(wrapper.find('.fa-volume-off')).toHaveLength(0);
    expect(wrapper.find('.fa-volume-up')).toHaveLength(1);
    expect(wrapper.find('#mute').filter("[title='Mute']")).toHaveLength(1);
    expect(wrapper.find('.fas').filter("[title='Mute']")).toHaveLength(1);
    
    wrapper.setProps({ isMuted : true });
    
    expect(wrapper.find('.fa-volume-off')).toHaveLength(1);
    expect(wrapper.find('.fa-volume-up')).toHaveLength(0);
    expect(wrapper.find('#mute').filter("[title='Unmute']")).toHaveLength(1);
    expect(wrapper.find('.fas').filter("[title='Unmute']")).toHaveLength(1);
  });
});

describe('click events', () => {
  it('calls handleStartStop when start_stop button clicked', () => {
    const mockHandleStartStop = jest.fn();
    const wrapper = shallow(<Controls {...props} handleStartStop={mockHandleStartStop}/>);
    wrapper.find('#start_stop').simulate('click');
    expect(mockHandleStartStop).toBeCalled();
  });

  it('calls reset when reset button clicked', () => {
    const mockReset = jest.fn();
    const wrapper = shallow(<Controls {...props} reset={mockReset} />);
    wrapper.find('#reset').simulate('click');
    expect(mockReset).toBeCalled();
  });

  it('calls handleMute when mute button clicked', () => {
    const mockMute = jest.fn();
    const wrapper = shallow(<Controls {...props} handleMute={mockMute} />);
    wrapper.find('#mute').simulate('click');
    expect(mockMute).toBeCalled();
  });
});

describe('snapshot', () => {
  it('renders', () => {
    const component = renderer.create(<Controls {...props}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});