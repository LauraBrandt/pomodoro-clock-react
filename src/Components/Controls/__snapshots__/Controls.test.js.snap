// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="controls"
>
  <button
    id="start_stop"
    onClick={[Function]}
    title="Start"
  >
    <i
      className="fas fa-play"
      title="Start"
    />
  </button>
  <button
    id="reset"
    onClick={[Function]}
    title="Reset"
  >
    <i
      className="fas fa-sync-alt"
      title="reset"
    />
  </button>
  <button
    id="mute"
    onClick={[Function]}
    title="Mute"
  >
    <i
      className="fas fa-volume-up"
      title="Mute"
    />
  </button>
</div>
`;
